gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/engine/access/index/GfxdIndexManager.java

```diff
@@ -438,8 +416,15 @@ public void onEvent(final LocalRegion owner, final EntryEventImpl event,
               owner.getFullPath());
         }
 
+        boolean isRecovered = false;
+        if (!owner.isInitialized()) {
+          DiskRegion dr = owner.getDiskRegion();
+          if (dr != null) {
+            isRecovered = dr.testIsRecovered(entry, false);
+          }
+        }
         boolean throwEEE = false;
-        if (!posDup && owner.isInitialized()) {
+        if (!posDup && (owner.isInitialized() || (!owner.isInitialized() && !isRecovered))) {
```

Comment:
Don't understand a few things here:

a) Without this additional check earlier code will fall into else and return silently without throwing an EEE. I think the expectation is that other initialized copies (e.g. GII source) will throw a proper exception. So what is this check achieving?

b) Don't quite get why it is appropriate to have this check for "!isRecovered", but not for "isRecovered". Can you add some more explanation of what exactly is going on here?